### PR TITLE
Adds LOOC.

### PR DESCRIPTION
**Pull request by FlattestGuitar**
- You can toggle whether you see it.
- Admins can toggle whether it's enabled.
- Config toggles for LOOC being enabled during rounds.
- Using L or Ctrl+L is a hotkey for LOOC because hawtkeyz.

Sorry for the stuff in `skin.dmf`, Atom got out of hand and this would've happened at some point anyway.

### CODE REVIEW

File: code/modules/client/verbs/ooc.dm [hunk 1]
```diff
@@ -80,6 +80,106 @@
...
+		if(M.client.prefs.chat_toggles & CHAT_LOOC)
+			clients_to_hear += M.client
+		for(var/mob/mob in M.client_mobs_in_contents)
+			if(mob.client.prefs.chat_toggles & CHAT_LOOC)
+				clients_to_hear += mob.client
+
+	for(var/client/C in GLOB.clients)
+		if(C in GLOB.admins)
+			if(C in clients_to_hear)
+				to_chat(C, "<span class='looc'><font color=LOOC_COLOR'>LOOC: [ADMIN_LOOKUPFLW(src)]: [msg]</font></span>")
```
Comment (FlattestGuitar) [inline code/modules/client/verbs/ooc.dm]: `span class='looc'` doesn't color the text, for some reason. Would love to help figuring out why.

**Comment by EldritchSigma** [comment]
I ded pls nerf

File: code/modules/client/verbs/ooc.dm [hunk 2]
```diff
@@ -80,6 +80,106 @@
...
+
+	if((copytext(msg, 1, 2) in list(".",";",":","#")) || (findtext(lowertext(copytext(msg, 1, 5)), "say")))
+		if(alert("Your message \"[raw_msg]\" looks like it was meant for in game communication, say it in LOOC?", "Meant for OOC?", "No", "Yes") != "Yes")
+			return
+
+	if(!holder)
+		if(handle_spam_prevention(msg, MUTE_LOOC))
+			return
+		if(findtext(msg, "byond://"))
+			to_chat(src, "<B>Advertising other servers is not allowed.</B>")
```
Comment (SamHPurp) [inline code/modules/client/verbs/ooc.dm]: span tags!

File: code/modules/client/verbs/ooc.dm (hunk 1 above)
Comment (SamHPurp) [inline code/modules/client/verbs/ooc.dm]: no `font` please... handle in the span

File: code/modules/client/verbs/ooc.dm [hunk 3]
```diff
@@ -80,6 +80,106 @@
...
+		for(var/mob/mob in M.client_mobs_in_contents)
+			if(mob.client.prefs.chat_toggles & CHAT_LOOC)
+				clients_to_hear += mob.client
+
+	for(var/client/C in GLOB.clients)
+		if(C in GLOB.admins)
+			if(C in clients_to_hear)
+				to_chat(C, "<span class='looc'><font color=LOOC_COLOR'>LOOC: [ADMIN_LOOKUPFLW(src)]: [msg]</font></span>")
+			else
+				to_chat(C, "<span class='looc'><font color='black'>(R)</font><font color='LOOC_COLOR'>LOOC: [ADMIN_LOOKUPFLW(src)]: [msg]</font></span>")
```
Comment (SamHPurp) [inline code/modules/client/verbs/ooc.dm]: no `font` please... handle in the span

File: code/modules/client/verbs/ooc.dm [hunk 4]
```diff
@@ -80,6 +80,106 @@
...
+		else if(C in clients_to_hear)
+			to_chat(C, "<span class='looc'><font color='LOOC_COLOR'>LOOC: [mob.name]: [msg]</font></span>")
```
Comment (SamHPurp) [inline code/modules/client/verbs/ooc.dm]: no `font` please... handle in the span

File: code/modules/client/verbs/ooc.dm [hunk 5]
```diff
@@ -80,6 +80,106 @@
...
+
+/proc/toggle_looc(toggle = null)
+	if(toggle != null)
+		if(toggle != GLOB.looc_allowed)
+			GLOB.ooc_allowed = toggle
+		else
+			return
+	else
+		GLOB.looc_allowed = !GLOB.looc_allowed
+	to_chat(world, "<B>The LOOC channel has been globally [GLOB.looc_allowed ? "enabled" : "disabled"].</B>")
```
Comment (SamHPurp) [inline code/modules/client/verbs/ooc.dm]: Use spans

File: code/modules/client/verbs/ooc.dm (hunk 1 above)
Comment (Birdtalon) [reply code/modules/client/verbs/ooc.dm]: Tested and this is down to BYOND caching the .css files. Remove the `<font>` tags and clear your cache and you'll be 🆗

**Comment by Lady-Luck** [comment]
After discussing this internally we've come to decision that ur gay

**Comment by AndrewMontagne** [comment]
This will be good to go as soon as the <b> tags have been replaced with appropriate spans.

**Pull request merged**